internal/storage/store.go

```diff
@@ -92,23 +93,35 @@ func NewFromConf(ctx context.Context, confWrapper *config.Wrapper) (Store, error
 
 // Store is the common interface implemented by storage backends.
 type Store interface {
-	Subscribable
 	// Driver is the name of the storage backend implementation.
 	Driver() string
-	// GetCompilationUnits gets the compilation units for the given module IDs.
-	GetCompilationUnits(context.Context, ...namer.ModuleID) (map[namer.ModuleID]*policy.CompilationUnit, error)
-	// GetDependents returns the dependents of the given modules.
-	GetDependents(context.Context, ...namer.ModuleID) (map[namer.ModuleID][]namer.ModuleID, error)
 	// ListPolicyIDs returns the policy IDs in the store
 	ListPolicyIDs(context.Context) ([]string, error)
 	// ListSchemaIDs returns the schema ids in the store
 	ListSchemaIDs(context.Context) ([]string, error)
 	// LoadSchema loads the given schema from the store.
 	LoadSchema(context.Context, string) (io.ReadCloser, error)
+}
+
+// SourceStore is implemented by stores that have policies in their source format (uncompiled).
+type SourceStore interface {
+	Store
+	Subscribable
+	// GetCompilationUnits gets the compilation units for the given module IDs.
+	GetCompilationUnits(context.Context, ...namer.ModuleID) (map[namer.ModuleID]*policy.CompilationUnit, error)
+	// GetDependents returns the dependents of the given modules.
+	GetDependents(context.Context, ...namer.ModuleID) (map[namer.ModuleID][]namer.ModuleID, error)
 	// LoadPolicy loads the given policy from the store
 	LoadPolicy(context.Context, ...string) ([]*policy.Wrapper, error)
 }
 
+// BinaryStore is implemented by stores that have pre-compiled policies in binary format.
+type BinaryStore interface {
```

Comment:
AFAIU the `BinaryStore` interface is used only as a policy loader, never as a store (binary store or not). Is it intended for future use?

Reply:
It embeds `Store` so it's a store as well. It's just that the scope of `Store` has been reduced to include only the fundamental methods. Other "helper" methods are now in specialised interfaces.